store/watcher_hub.go

```diff
@@ -61,7 +63,9 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index, storeInde
 	}
 
 	w := &watcher{
-		eventChan:  make(chan *Event, 100), // use a buffered channel
+		// The eventChan needs to be at least as big as the event history to support streaming
+		// events from an outdated index.
+		eventChan:  make(chan *Event, wh.EventHistory.Queue.Capacity),
```

Comment:
I was thinking about that. In theory this queue should be mostly empty most of the time and ~5Mb overhead should be Ok (we have about 500 watchers in a 100 node cluster).